packages/app-project/src/shared/components/SubjectPicker/helpers/checkRetiredStatus.js

```diff
@@ -0,0 +1,20 @@
+import { panoptes } from '@zooniverse/panoptes-js'
+const env = 'production'
+
+export default async function checkRetiredStatus(subject_ids, workflow, page_size=20) {
+  const workflow_id = workflow.id
+  const retirementStatuses = {}
+  const response = await panoptes
+    .get('/subject_workflow_statuses', {
+      env,
+      page_size,
+      subject_ids,
+      workflow_id
+    })
+  const { subject_workflow_statuses } = response.body
+  subject_workflow_statuses.forEach(status => {
+    const inProgress = status.classifications_count > 0 ? 'In progress' : 'Unclassified'
+    retirementStatuses[status.links.subject] = status.retired_at ? 'Retired' : inProgress
+  })
+  return retirementStatuses
```

Comment:
⚠️ Let me talk myself through some sanity checks here.

OK, so on a _successful_ scenario, we get data from /subject_workflow_statuses, and we use the data from that to populate 'retirementStatuses' (an initially empty object `{}`), which is returned when `fetchRows()` calls `const retirementStatuses = await checkRetiredStatus(subject_ids, workflow, page_size)`. Cool!

⚠️ For failure scenarios, here's what can possibly go wrong:
1. panoptes doesn't return a valid result (e.g. endpoint is dead) and returns a bad response. (It probably won't just return undefined.)
2. Will a bad `response` have .body? And will that .body have .subject_workflow_statuses? (Probably not)
3. If subject_workflow_statuses is _undefined,_ that'll throw an error out of `checkRetiredStatus()` and send it up the stack to fetchRows, which can't handle the error either.

Thoughts: `checkRetiredStatus` seems a bit brittle. If it breaks, it'll break everything downstream. Maybe wrap everything in a try-catch so it always returns an empty object `{}` on error?

Counterpoint: Wait, Shaun, do you REALLY want a try-catch to _hide_ the error? If getting the value of subject_workflow_statuses from panoptes _is essential,_ then _it's best_ that an error at checkRetiredStatus() ends up crashing everything downstream, so we know something went wrong at panoptes. Better an explicit error than leading users around with false values.

Counter-counter point: OK, fine, go ahead and crash, but at least spit out an error message that makes sense to users.

Reply:
There's an issue with Panoptes where connections will occasionally reset during a query. See https://github.com/zooniverse/panoptes-cli/issues/193

I think you're right that API requests should be wrapped in try/catch. I'll add Sentry logging too.

Reply:
👍